### ISSUE TITLE
Lots of regression in exit codes [from 1.12 to master]rr

### ISSUE THREAD

**Issue by runcom** [docker/docker#26152]
Under docker-1.10, flag.go exits 125 when a command-line flag is missing a required argument. With docker 1.12 *most* commands do so... except for a seemingly random few:
```
   # docker load --input;echo $?
   flag needs an argument: --input
   See 'docker load --help'.
   1    <-- 125 with docker-1.10

   # docker ps --filters;echo $?         <-- same with -n or --format
   unknown flag: --filters
   See 'docker ps --help'.
   1    <-- 125 with docker-1.10
```
(List is not meant to be exhaustive; there are more).

If my diagnosis is correct, root cause is that as part of the cobra migration in June some NewXxxCommand() functions were added with this line:
```
    cmd.SetFlagErrorFunc(flagErrorFunc)
```
...and some were not:
```
    $ grep -L lagErrorFunc api/client/{image,container}/* | wc -l
    23
```
If this analysis is correct, I hope there's a less duplicationy way to solve this than adding cmd.SetFlagErrorFunc() to each source file!

@vdemeester @dnephin

**Comment by dnephin** [comment]
`SetFlagErrorFunc()` should only be necessary on the root command, not every command. I removed it from all the commands in #25354

I just tried the examples with a binary built on master and they all return the correct `125`. Can you try master and see if it's fixed?

**Comment by runcom** [comment]
@dnephin tried master, works fine, closing (thx)

**Comment by thaJeztah** [comment]
ping @dnephin @vdemeester @runcom I checked this issue with 1.11, and it looks like this was a regression in 1.12. Do you think it's possible (and worth it) to fix just this issue for 1.12.2 (without including the conversion of `docker` to Cobra)?

**Comment by thaJeztah** [comment]
Also ping @vieux ^^

**Comment by runcom** [comment]
> ping @dnephin @vdemeester @runcom I checked this issue with 1.11, and it looks like this was a regression in 1.12. Do you think it's possible (and worth it) to fix just this issue for 1.12.2 (without including the conversion of docker to Cobra)?

this is huge and I don't have any clue if we can fix just this for 1.12.2 - would be great though if anyone can do this fast for 1.12.2 :)

**Comment by thaJeztah** [comment]
@runcom I opened https://github.com/docker/docker/issues/26775 so that we don't loose sight of the issue
